src/main/java/net/mcreator/element/parts/gui/InputSlot.java

```diff
@@ -19,25 +19,28 @@
 package net.mcreator.element.parts.gui;
 
 import net.mcreator.element.parts.MItemBlock;
+import net.mcreator.element.parts.procedure.LogicProcedure;
 import net.mcreator.element.parts.procedure.Procedure;
 
 import java.awt.*;
 
 public class InputSlot extends Slot {
 
 	public MItemBlock inputLimit;
+	public LogicProcedure disablePlacement;
 
 	// for deserialization use only, to specify default values
 	@SuppressWarnings("unused") private InputSlot() {
 		this.dropItemsWhenNotBound = true;
 	}
 
-	public InputSlot(int id, int x, int y, Color color, boolean canTakeStack,
-			boolean dropItemsWhenNotBound, Procedure onSlotChanged, Procedure onTakenFromSlot,
-			Procedure onStackTransfer, MItemBlock inputLimit) {
-		super(id, x, y, color, canTakeStack, dropItemsWhenNotBound, onSlotChanged, onTakenFromSlot,
-				onStackTransfer);
+	public InputSlot(int id, int x, int y, Color color, LogicProcedure disablePickupCondition,

```

Comment:
disablePickupCondition should be renamed to disablePickup to keep the naming consistent